### PR TITLE
[ENH] Add options to control ICA attempts

### PR DESCRIPTION
**Pull request by tsalo**
References #218.

Changes proposed in this pull request:

- Add options for the maximum number of iterations and restarts in ICA.
- Remove `fixed_seed` argument for `io.writeresults` and stop returning it in `tedica`.

### CODE REVIEW

**Comment by emdupre** [comment]
This is amazing, thank you ! One question: Should `maxit` be 500 or 5000 ?

**Comment by tsalo** [comment]
I'd like to drop it down to 500 (although I see now that I accidentally left it as 5000 in the tedica docstring).
Once we've merged this, though, I am planning to expand the reliability analysis so that we also look at how many iterations it generally takes to converge, as well as how many restarts we'll generally need (in conjunction with a reasonable number of iterations per restart) to have a strong chance of reaching convergence. I think we can come up with better defaults than 500 and 5 from that analysis.

**Comment by emdupre** [comment]
Having this as a point for the reliability analysis seems like a great idea :+1: 

Just looking through our existing integration tests, it looks like the seed-updating works nicely, but isn't capped at 5 re-starts.

Three-echo:
```
INFO:tedana.decomposition.eigendecomp:ICA attempt 1 converged in 96 iterations
```

Five-echo
```
WARNING:tedana.decomposition.eigendecomp:ICA attempt 1 failed to converge after 500 iterations
WARNING:tedana.decomposition.eigendecomp:Random seed updated to 43
WARNING:tedana.decomposition.eigendecomp:ICA attempt 2 failed to converge after 500 iterations
WARNING:tedana.decomposition.eigendecomp:Random seed updated to 44
WARNING:tedana.decomposition.eigendecomp:ICA attempt 3 failed to converge after 500 iterations
WARNING:tedana.decomposition.eigendecomp:Random seed updated to 45
WARNING:tedana.decomposition.eigendecomp:ICA attempt 4 failed to converge after 500 iterations
WARNING:tedana.decomposition.eigendecomp:Random seed updated to 46
WARNING:tedana.decomposition.eigendecomp:ICA attempt 5 failed to converge after 500 iterations
WARNING:tedana.decomposition.eigendecomp:Random seed updated to 47
WARNING:tedana.decomposition.eigendecomp:ICA attempt 6 failed to converge after 500 iterations
WARNING:tedana.decomposition.eigendecomp:Random seed updated to 48
INFO:tedana.decomposition.eigendecomp:ICA attempt 7 converged in 146 iterations
```

File: tedana/decomposition/eigendecomp.py [hunk 1]
```diff
@@ -423,20 +429,28 @@ def tedica(n_components, dd, fixed_seed):
...
+                LGR.warning('Random seed updated to {0}'.format(fixed_seed))
+            else:
+                LGR.info('ICA attempt {0} converged in {1} '
+                         'iterations'.format(i_attempt + 1, ica.n_iter_))
+                break
 
     mmix = ica.mixing_
     mmix = stats.zscore(mmix, axis=0)
-    return mmix, fixed_seed
+    return mmix
```
Comment (emdupre) [inline tedana/decomposition/eigendecomp.py]: Is there a reason to not return the `fixed_seed` value, now ? I think it'd still be nice to have to keep provenance, but maybe there's a reason to remove it that I'm missing....

Comment (tsalo) [reply tedana/decomposition/eigendecomp.py]: I think we must have originally used it in `io.writeresults` for something (filenames maybe?), but that function no longer uses the seed so it's not actually used anywhere. Plus, the log file will now have that info, along with number of iterations required for convergence, for those who might want it.

**Comment by emdupre** [comment]
Ah, I'm an idiot, you have `maxrestart 10` in the test call. Should we just set that as the default ? Or was there a particular reason you used it there ?

**Comment by tsalo** [comment]
I went with 5 for a default because I thought that was what MELODIC had (turns out it's 6), but would be happy to change to 10 to match the tests. I just chose 10 in the five-echo test because I knew convergence wasn't happening with 5 unless I chose a specific seed. Would you prefer 10 as our default? There's a nice symmetry to it since with the default `maxit` being 500 that would be equal to a cap of 5000 iterations (our old default).

File: tedana/decomposition/eigendecomp.py (hunk 1 above)
Comment (emdupre) [reply tedana/decomposition/eigendecomp.py]: We did use it, I think to go in the summary file. But if we're ok that it's not just in the log file, that sounds fine by me ! :+1:

**Comment by emdupre** [comment]
I think I'd prefer 10 as a default, since then at least all of our test datasets pass under our default params _and_ it has that symmetry :sparkles: If that sounds reasonable to you !

**Comment by tsalo** [comment]
👍 That works for me. Hopefully we'll be able to come up with more robust defaults from the reliability analysis, but I'm quite happy with 500 and 10.

File: .circleci/config.yml [hunk 2]
```diff
@@ -198,7 +198,7 @@ jobs:
             source activate venv
             cd /tmp/data/five-echo/
             tedana -d p06.SBJ01_S09_Task11_e[1,2,3,4,5].sm.nii.gz \
-                -e 15.4 29.7 44.0 58.3 72.6 --verbose \
+                -e 15.4 29.7 44.0 58.3 72.6 --verbose --maxit 500 --maxrestart 10 \
```
Comment (emdupre) [inline .circleci/config.yml]: Can we remove these extra arguments since they're the defaults ?

**Comment by emdupre** [comment]
One small question, but this looks great ! Excited to have this merged !! :rocket:

File: .circleci/config.yml (hunk 2 above)
Comment (tsalo) [reply .circleci/config.yml]: 👍

**Comment by emdupre** [comment]
LGTM ! Merging :sparkles: :rocket:

**Pull request merged**